### ISSUE TITLE
RippleColor via Service

### ISSUE THREAD

**Issue by flodaniel** [edusperoni/nativescript-ng-ripple#7]
So we are setting the the rippleColor with a service. We do this in an effort to make all colors in the app changeable by fetching the relevant colors from our server.
This is how we would want to use this for example on a StackLayout
   ` <StackLayout ripple [rippleColor]="this.themeService.getCSSVariable('--grey-mid-light')">`

However the Angular change detection on the components run really early, and at this poin tht themeService does only return a placeholder color (which would be fine). but by the time we detect in in the `getCssVariable` method, that we are not ready, change detection has run the component throws an error. 
To explain:
This is the implementation of getCssVariable(name: string): string

```
    public getCSSVariable(name: string): string {
        const view = this.getTopUIFrame();
        if (!view || !view.style) {
            console.log("returning early");

            return '#e8f9ee';
        }

        return view.style.getCssVariable(name);
    }
```

However this leads to the following error

`ERROR TypeError: Cannot read property 'currentValue' of undefined
`
Changing the method to the following would work (but would defeat the purpose):

```
   public getCSSVariable(name: string): string {
        return '#e8f9ee';

        const view = this.getTopUIFrame();
        if (!view || !view.style) {
            console.log("returning early");

            return '#e8f9ee';
        }

        return view.style.getCssVariable(name);
    }
```

The error occurs in the change detection of the plugin in the file ng-ripple.directive.js on line 45:

```
    NativeRippleDirective.prototype.ngOnChanges = function (changes) {
        if (changes.ripple.currentValue !== changes.ripple.previousValue ||
            changes.rippleColor.currentValue !== changes.rippleColor.currentValue) {
            this.applyOrRemoveRipple();
        }
    };
```
Because currentValue is accessed without checking, which changes occured and if we have a value in the change.

Is there a way around this, that I am not seeing?
I would propose a simple change by adding a check for `changes.ripple` to the if-clause: 

```
        if (changes.ripple && changes.ripple.currentValue !== changes.ripple.previousValue ||
            changes.rippleColor.currentValue !== changes.rippleColor.currentValue) {
            this.applyOrRemoveRipple();
        }

```

Any help is appreciated :)

**Comment by flodaniel** [comment]
So there is a beta release (3.0.0-beta.1) which does not have this issue. The source code is completly refactored in that beta version. 
However the last commit is from 22.05.2019 on the beta branch. So it seems that development has stopped and the plugin is not maintained any more.

**Comment by edusperoni** [comment]
Fixed via #8 and beta version
